esp/services/ws_codesign/ws_codesignService.cpp

```diff
@@ -47,17 +47,18 @@ bool Cws_codesignEx::onSign(IEspContext &context, IEspSignRequest &req, IEspSign
 {
     resp.setRetCode(-1);
 
-    StringBuffer keyid(req.getKeyIdentifier());
-    keyid.trim();
+    StringBuffer userid(req.getUserID());
```

Comment:
The userid field needs to be XML decoded to make sure any characters are decoded correctly.

Reply:
ESP lower layer already did the decoding, so the value here is the xml-decoded value.